test/m_f_c_coin.js

```diff
@@ -18,5 +84,350 @@ contract('MFC_Token', function(accounts) {
+            token_contract = instance;
+            return token_contract.transfer(user, 1, {'from': owner});
+        }).then(function() {
+            return token_contract.burn(10, {'from': user});
+        }).catch(function(e){
+            assert.equal(e, 'Error: VM Exception while processing transaction: revert', "something went wrong");
+        });
+    });
+
+    /* burning without balan*ce */
+    it("the second account does not have a record in the balance sheet and will try to burn 10 tokens and should receive an error: revert", function() {
+        var user = accounts[3];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.burn(10, {'from': user});
+        }).catch(function(e){
+            assert.equal(e, 'Error: VM Exception while processing transaction: revert', "something went wrong");
+        });
+    });
+
+    /* balance of burner = 0 */
+    it("the second account have a record in the balance sheet but have 0 tokens and will try to burn 10 tokens and should receive an error: revert", function() {
+        var owner = accounts[0];
+        var user = accounts[4];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.transfer(user, 1, {'from': owner});
+        }).then(function() {
+            return token_contract.burn(1, {'from': user});
+        }).then(function() {
+            return token_contract.burn(10, {'from': user});
+        }).catch(function(e){
+            assert.equal(e, 'Error: VM Exception while processing transaction: revert', "something went wrong");
+        });
+    });
+});
+
+contract('MFC_Token tests burnAll', async function(accounts) {
+
+    /* Task 18 - Create test for MFC_coin burnAll() with Truffle */
+
+    /* Using Truffle, we check method for MFC_coin burnAll() and test 3 variants:
+       - burner have balance */
+    it("should transfer 10 tokens to second account and second account should burn all tokens", function() {
+        var user_balance_befor;
+        var owner = accounts[0];
+        var user = accounts[5];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.transfer(user, 10, {'from': owner});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            user_balance_befor = balance.valueOf();
+            return token_contract.burnAll({'from': user});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            return balance.valueOf();
+        }).then(function(user_balance_after){
+            assert.equal(user_balance_befor, 10, "something went wrong");
```

Comment:
It is needed to write, what exactly went wrong in all assert cases.
Also, you haven't checked that if a user has 10 tokens and try to burn not all of them but 3, he should found that after that he has 7 tokens.